### PR TITLE
Use zeitgeist for dependency verification

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:

We now use a dedicated job for verifying our dependencies while pinning them in `dependencies.yaml`.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

None

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -54,12 +60,18 @@ integration-static: .install.ginkgo # It needs to be release so we correctly tes
 	sudo -E "$(GOTOOLS_BINDIR)/ginkgo" $(TEST_FLAGS) $(GINKGO_FLAGS)
 
 .install.ginkgo:
-	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go install github.com/onsi/ginkgo/v2/ginkgo@latest
+	GOBIN=$(abspath $(GOTOOLS_BINDIR)) go install github.com/onsi/ginkgo/v2/ginkgo@v$(GINKGO_VERSION)
```
Comment (rphillips) [inline Makefile:63]: Let's move the 'v' into the GINKGO_VERSION, so there isn't any 'magic'.

File: Makefile [hunk 2]
```diff
@@ -54,12 +60,18 @@ integration-static: .install.ginkgo # It needs to be release so we correctly tes
...
 
 .install.golangci-lint:
 	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | \
 		BINDIR=$(abspath $(GOTOOLS_BINDIR)) sh -s $(GOLANGCI_LINT_VERSION)
 
+$(GOTOOLS_BINDIR)/zeitgeist:
+	mkdir -p $(GOTOOLS_BINDIR)
+	curl -sSfL -o $(GOTOOLS_BINDIR)/zeitgeist \
+		https://github.com/kubernetes-sigs/zeitgeist/releases/download/v$(ZEITGEIST_VERSION)/zeitgeist_$(ZEITGEIST_VERSION)_linux_amd64
```
Comment (rphillips) [inline Makefile:72]: ditto

File: dependencies.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  - name: go
+    version: 1.19
+    refPaths:
+      - path: .github/workflows/ci.yml
+        match: GO_VERSION
+      - path: .github/workflows/release.yml
+        match: GO_VERSION
+
+  - name: rust
+    version: 1.58.1
```
Comment (rphillips) [inline dependencies.yaml:12]: Can we use a newer version of rust?

File: Makefile (hunk 2 above)
Comment (saschagrunert) [reply Makefile]: Fixed :+1:

File: Makefile (hunk 1 above)
Comment (saschagrunert) [reply Makefile]: Done :+1:

File: dependencies.yaml (hunk 3 above)
Comment (saschagrunert) [reply dependencies.yaml:12]: I don't think so. Maybe @lsm5 has more input on when to expect new rust versions on RHEL / RHCOS 8/9.

**Pull request merged**